frontend/src/components/common/ApiErrorBoundary/index.tsx

```diff
@@ -0,0 +1,48 @@
+import ApiError from '@api/ApiError';
+import { QueryErrorResetBoundary } from '@tanstack/react-query';
+import { ErrorBoundary, ErrorBoundaryPropsWithComponent, FallbackProps } from 'react-error-boundary';
+import * as Sentry from '@sentry/react';
+
+interface ApiErrorBoundaryProps extends ErrorBoundaryPropsWithComponent {
+  FallbackComponent: React.ComponentType<FallbackProps>;
+}
+
+export default function ApiErrorBoundary({ FallbackComponent, children }: ApiErrorBoundaryProps) {
+  const fallbackRender = (props: FallbackProps) => {
+    const error = props.error as ApiError;
+    if (!(error instanceof ApiError)) {
+      throw error;
+    }
```

Comment:
앞에서 `ApiError`로의 타입 단언을 사용하여 `error`를 선언했는데, 그 `error`에 대해 바로 다음 줄에서 `ApiError` 인스턴스인지를 체크하는 것은 조금 모순된 로직이 아닌가 싶습니다. `props.error`가 `ApiError` 인스턴스가 아닌 경우에 대한 처리를 명시적으로 진행한 다음에 `error`를 선언하면 어떨지 제안드려 봅니다.

```suggestion
    if (!(props.error instanceof ApiError)) {
      throw props.error;
    }
    const error = props.error;
```

Reply:
TS에서 타입 단언이 타입 변환을 의미하진 않는 것으로 알고 있습니다. 단언은 개발자 편의를 위한 것일 뿐 런타임 시 타입 평가에 영향을 미치진 않아요.

> Like a type annotation, type assertions are removed by the compiler and won’t affect the runtime behavior of your code.
> [Type Assertions](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-assertions)

하지만 아르가 말씀하신 것 처럼 혼란을 불러올만한 코드인 것 같아서 순서를 바꿔보도록 하겠습니다!
피드백 감사해요.👍